### DISCUSSION TITLE
Disable flow from homeassistant

### DISCUSSION THREAD

**Discussion by MrJoki007** [zachowj/hass-node-red#207] [Q&A]
Hello,
im using this addon to send values from nodered to homeassistant.
That works perfect.

Now i want to integrate a switch in homeassistant to disable one flow.
How can i do that?

In the description of this addon it says: Disable and enable Node-RED flows from Home Assistant UI

**Comment by zachowj** [comment]
Under one of the event nodes at the bottom of the config, there should be a checkbox "Expose to Home Assistant". When this is checked, it will create a switch entity in Home Assistant. When this switch is turned off, the node will no longer output.

https://zachowj.github.io/node-red-contrib-home-assistant-websocket/guide/custom_integration/exposed-nodes.html

**Reply by starman2k01** [reply]
That's not what he asked for. He (and I) want to expose an entire flow to home assistant and disable it. The only way to do that now is to crowbar a switch to enable/disable various nodes that are in flows. Why can't we simply disable an entire flow? Some flows are extremely complex with multiple trigger nodes!

**Reply by tempotellergabel** [reply]
I have a similar issue with this function. For some Flows I have multiples inputs (switches, remotes, motion sensors, etc.). So I would need to expose each input node. That results in multiple toggles which doesn't make sense for my use case.


So now I use "Helper", which I set up in Home Assistant and add to the flow using the Node current_state
It's working but it's not that elegant since I need to add a current_state node after each input node. The advantage is that it is visualized in the flow overview, which I much prefer over some box kind of hidden in the UI of the input node. (It's actually the main reason I still prefer Node-RED, that the flows are nicely visualized)


**However, I would suggest to adjust the description on the Github page which currently says:**

> Key Feature:
> Flow Control:
> Enable or disable **Node-RED flows** directly from the Home Assistant UI.

**Reply by zolakt** [reply]
I agree completely. I'm also looking for a way to enable/disable flows programatically, and that documentation is really missleading.

For me the suggested workaround, apart from being really cumbersome to maintain, are not the solution, since those input triggers still run all the time. You just reject the message at the next block. Disabling the flow stops the triggers as well.

Currently I just do this manually for most flows, since I need to enable/disable just one a year (summer/winter).
For the ones I go need to trigger more often, and on an irregular basis (e.g. vaction mode, having guests etc), I do that the same way as you guys suggested. A helper toggle in HA, and in each flow, after each trigger, I check the state of that toggle. It's far from an ideal solution, but the only way I managed to get any functionality
